packages/twenty-server/src/engine/metadata-modules/agent/agent-chat.controller.ts

```diff
@@ -0,0 +1,65 @@
+import {
+  Body,
+  Controller,
+  Get,
+  Param,
+  Post,
+  Res,
+  UseFilters,
+  UseGuards,
+} from '@nestjs/common';
+
+import { Response } from 'express';
+
+import { RestApiExceptionFilter } from 'src/engine/api/rest/rest-api-exception.filter';
+import { JwtAuthGuard } from 'src/engine/guards/jwt-auth.guard';
+import { WorkspaceAuthGuard } from 'src/engine/guards/workspace-auth.guard';
+
+import { AgentChatService } from './agent-chat.service';
+import { AgentStreamingService } from './agent-streaming.service';
+
+@Controller('rest/agent-chat')
+@UseGuards(JwtAuthGuard, WorkspaceAuthGuard)
+@UseFilters(RestApiExceptionFilter)
+export class AgentChatController {
+  constructor(
+    private readonly agentChatService: AgentChatService,
+    private readonly agentStreamingService: AgentStreamingService,
+  ) {}
+
+  @Get('threads/:agentId')
+  async getThreadsForAgent(@Param('agentId') agentId: string) {
+    return this.agentChatService.getThreadsForAgent(agentId);
+  }
+
+  @Get('messages/:threadId')
```

Comment:
This doesn't feel like the most restful way to write this, it should rather be /threads/{id}/messages I think